pkg/logging/journald_logger.go

```diff
@@ -97,23 +98,32 @@ func (journaldLogger *JournaldLogger) Process(dataStore string, config *logging.
 	vars := map[string]string{
 		"SYSLOG_IDENTIFIER": syslogIdentifier,
 	}
-	var wg sync.WaitGroup
-	wg.Add(2)
-	f := func(wg *sync.WaitGroup, r io.Reader, pri journal.Priority, vars map[string]string) {
+	journaldLogger.vars = vars
+	return nil
+}
+
+func (journaldLogger *JournaldLogger) Process(wg *sync.WaitGroup, stdout chan string, stderr chan string) error {
+	defer wg.Done()
+	var internalWg sync.WaitGroup
+	internalWg.Add(2)
+	f := func(wg *sync.WaitGroup, dataChan chan string, pri journal.Priority, vars map[string]string) {
 		defer wg.Done()
-		s := bufio.NewScanner(r)
-		for s.Scan() {
-			if s.Err() != nil {
-				return
+		for {
+			select {
+			case log, ok := <-dataChan:
+				if ok {
+					journal.Send(log, pri, vars)
+				} else {
+					return
+				}
 			}
-			journal.Send(s.Text(), pri, vars)
 		}
 	}
 	// forward both stdout and stderr to the journal
-	go f(&wg, config.Stdout, journal.PriInfo, vars)
-	go f(&wg, config.Stderr, journal.PriErr, vars)
+	go f(&internalWg, stdout, journal.PriInfo, journaldLogger.vars)
```

Comment:
Passing WaitGroup into functions is not necessary IMO, as it can be used directly.